lib/tokenlist.cpp

```diff
@@ -2096,3 +2116,25 @@ bool TokenList::isKeyword(const std::string &str) const
     static const auto& latest_c_keywords = Keywords::getAll(Standards::cstd_t::CLatest);
     return latest_c_keywords.find(str) != latest_c_keywords.end();
 }
+
+bool TokenList::isC() const
+{
+    assert(mLang != Standards::Language::None);
```

Comment:
hmm .. do we need to disallow None? parsing project files / markup files is some cases when it makes sense to not require C/C++.

Reply:
Currently that is the best solution to make sure we do not have cases where it isn't specified. Also it was a binary state before so it was treated as either C++ or C based on the code (some adjustments to that coming after this and the already existing follow-up PRs have been merged) and never as "not source code".

I also think we do not need the tokenizer at all for the project files. I am sure that can be solved with much simpler approach. That parsing in the projects also seems to be incomplete but I will extend the testing with more examples later on.

The markup handling also needs more testing (as you pointed pout there might be some short comings). The current force is simple brute force and it should be more heuristic in nature to avoid false negatives. I am pretty sure with added test cases the picture will be much clearer.

These changes will mainly properly define how it being treated instead of differently based on the entry point or input. I do no expect this to have any effect for the end user at all.